packages/components/src/Card/Card.mdx

```diff
@@ -112,3 +114,72 @@ Card titles should be sentence-cased.
     </Content>
   </Card>
 </Playground>
+
+### Header with button
+
+<Playground>
+  <Card
+    header={{
+      title: "Banana varieties",
+      action: {
+        label: "Buy",
+        type: "primary",
+        icon: "add",
+        onClick: () => {
+          alert("Buy Bananas!");
+        },
+      },
+    }}
+  >
+    <Content>
+      <Heading level={4}>Burro</Heading>
+      <Text>
+        Squatty and slightly square at the edges, these bananas are slightly
+        tangy and lemony. When ripe, the fruit is soft and the skin is yellow
+        with black spots. Also called Horse, Hog, or Orinoco bananas.
+      </Text>
+      <Heading level={4}>Cavendish</Heading>
+      <Text>
+        This is the familiar yellow banana sold in United States’ supermarkets.
+        There are also other sizes of Cavendish, including Dwarf and Giant,
+        though they are difficult to find.
+      </Text>
+    </Content>
+  </Card>
+</Playground>
+
+### Header with custom component
```

Comment:
Can we add some guidance around when to use a custom component? Something like "Use the provided header title and action properties when possible. If you require UI elements not supported by the card's header, a custom component can be passed in."